lib/git-shell-out-strategy.js

```diff
@@ -422,6 +422,39 @@ export default class GitShellOutStrategy {
     return this.exec(args, {writeOperation: true});
   }
 
+  async getCommitMessageTemplate() {
+    let templatePath = await this.getConfig('commit.template');
+    if (!templatePath) {
+      return null;
+    }
+
+    /**
+     * Get absolute path from git path
+     * https://git-scm.com/docs/git-config#git-config-pathname
+     */
+    const homeDir = os.homedir();
+    // this regex attempts to get the specified user's home directory
+    // Ex: on Mac ~kuychaco/ is expanded to the specified user’s home directory (/Users/kuychaco)
+
+    // Regex translation:
+    // ^~ line starts with tilde
+    // ([^/]*)/ captures non-forwardslash characters before first slash
+    const regex = new RegExp('^~([^/]*)/');
```

Comment:
We didn't really have a strong reason for preferring the regexp constructor over literals here.  We went with what the original author had.

Out of curiosity, we looked to see if there was a performance difference.  Answer: not really.
https://www.measurethat.net/Benchmarks/Show/1734/1/regexp-constructor-vs-literal

We did give it a more descriptive name, and moved it out of this function so as to avoid creating a new regex every time.